lib/registry.js

```diff
@@ -114,8 +113,11 @@ exports.load = function(registry) {
               checkRetry();
             }, timeout);
 
-            Promise.resolve()
-            .then(function() {
+            // in case registry is being reconfigured, chain on a promise
+            // which delivers the registry to use, when it is ready
+            (self.queue || Promise.resolve(self))
```

Comment:
Perhaps something that indicates this is private and describes what it is? `self.reconfigPromise_`?